embedding/csharp/Tizen.Flutter.Embedding/Plugins/FlutterDotnetPluginRegistrar.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+namespace Tizen.Flutter.Embedding
+{
+    internal class FlutterDotnetPluginRegistrar : IFlutterPluginRegistrar
+    {
+        private IPluginRegistry _registry;
+
+        public FlutterDotnetPluginRegistrar(IPluginRegistry registry)
+        {
+            _registry = registry;
+        }
+
+        public void RegisterPlugin(IFlutterPlugin plugin)
+        {
+            if (plugin != null)
+            {
+                FlutterDesktopPluginRegistrar registrar = _registry.GetRegistrarForPlugin(plugin.GetType().FullName);
```

Comment:
I agree, we can use the singleton `DefaultBinaryMessenger` using the engine's `FlutterDesktopMessenger` handle. Currently, only one registrar is used in `flutter_tizen.cc`, but in case the flutter registrar system changes in the future, I decided that it would be safe to use `FlutterDesktopMessenger` using the registrar obtained through `FlutterDesktopPluginRegistrarGetMessenger`.

```cpp
FlutterDesktopPluginRegistrarRef FlutterDesktopGetPluginRegistrar(
    FlutterDesktopEngineRef engine,
    const char* plugin_name) {
  // Currently, one registrar acts as the registrar for all plugins, so the
  // name is ignored. It is part of the API to reduce churn in the future when
  // aligning more closely with the Flutter registrar system.
  return EngineFromHandle(engine)->plugin_registrar();
}
```

Reply:
That plugin registrar (`FlutterDesktopPluginRegistrar`) is used for registering native plugins. You have your own registrar implementation (`FlutterDotnetPluginRegistrar`) for .NET plugins and it manages their lifecycles. The two registrars should be independent of each other.

Reply:
Thanks for kind explanation, `FlutterDesktopMessenger` of the engine can be used.